labcontrol/db/support_files/patches/python_patches/2.py

```diff
@@ -0,0 +1,90 @@
+from labcontrol.db.sql_connection import TRN
+import logging
+
+
+with TRN:
+    # although TRN has its own queue, use a local list to queue statements
+    # until prerequisite queries are completed.
+    statements = []
+
+    #
+    # rename kappa_hyper_plus_kit_id in library_prep_shotgun_process
+    #
+    sql = """SELECT COUNT(*) FROM information_schema.columns
+             WHERE table_name = 'library_prep_shotgun_process'
+             AND column_name = 'kappa_hyper_plus_kit_id';"""
+    logging.debug(sql)
+    TRN.add(sql)
+    result = TRN.execute_fetchflatten()[0]
+
+    # for now, assume result will be either 0 or 1.
+    # if 1, then the column 'kappa_hyper_plus_kit_id' needs
+    # to be renamed to 'kapa_hyper_plus_kit_id', otherwise
+    # it has already been renamed.
+    if result == 1:
+        msg = ('labcontrol.library_prep_shotgun_process contains the '
+               'column name kappa_hyper_plus_kit_id. Will rename to '
+               'kapa_hyper_plus_kit_id...')
+        logging.debug(msg)
+
+        sql = """ALTER TABLE labcontrol.library_prep_shotgun_process RENAME
+        COLUMN kappa_hyper_plus_kit_id to kapa_hyper_plus_kit_id;"""
+        statements.append(sql)
+    else:
+        msg = ('labcontrol.library_prep_shotgun_process contains the '
+               'column name kapa_hyper_plus_kit_id. Skipping...')
+        logging.debug(msg)
+
+    #
+    # rename fk_shotgun_library_prep_process_kappa in
+    # library_prep_shotgun_process
+    #
+    sql = """SELECT COUNT(*) FROM information_schema.constraint_column_usage
+                 WHERE table_name = 'library_prep_shotgun_process' AND
+                 constraint_name = 'fk_shotgun_library_prep_process_kappa';"""
+
+    logging.debug(sql)
+    TRN.add(sql)
+    result = TRN.execute_fetchflatten()[0]
```

Comment:
I wonder if `fetchflatten` has too strong assumption of the data structure. Maybe it's good to comment that the first column is always something.

Reply:
Thanks! Quite used to it from Qiita, but yeah it doesn't read well. I'll add a comment.